test/server/cards/02.3-ItFC/WarDogMaster.spec.js

```diff
@@ -0,0 +1,45 @@
+describe('War Dog Master', function() {
+    integration(function() {
+        describe('War Dog Master\'s ability', function() {
+            beforeEach(function() {
+                this.setupTest({
+                    phase: 'conflict',
+                    player1: {
+                        inPlay: ['war-dog-master'],
+                        dynastyDiscard: ['adept-of-the-waves', 'favorable-ground']
+                    },
+                    player2: {
+                    }
+                });
+
+                this.warDogMaster = this.player1.findCardByName('war-dog-master');
+                this.adeptOfTheWaves = this.player1.findCardByName('adept-of-the-waves', 'dynasty discard pile');
+                this.favorableGround = this.player1.findCardByName('favorable-ground', 'dynasty discard pile');
+            });
+
+            it('should work with a character', function() {
+                this.player1.moveCard(this.adeptOfTheWaves, 'dynasty deck');
+                this.noMoreActions();
+                this.initiateConflict({
+                    attackers: [this.warDogMaster]
+                });
+                expect(this.player1).toHavePrompt('Triggered Abilities');
+                this.player1.clickCard(this.warDogMaster);
+                expect(this.getChatLogs(2)).toContain('player1 uses War Dog Master, discarding Adept of the Waves to give War Dog Master +2military');
+                expect(this.warDogMaster.getMilitarySkill()).toBe(4);
+            });
+
+            it('should work with a holding', function() {
+                this.player1.moveCard(this.favorableGround, 'dynasty deck');
+                this.noMoreActions();
+                this.initiateConflict({
+                    attackers: [this.warDogMaster]
+                });
+                expect(this.player1).toHavePrompt('Triggered Abilities');
+                this.player1.clickCard(this.warDogMaster);
+                expect(this.getChatLogs(2)).toContain('player1 uses War Dog Master, discarding Favorable Ground to give War Dog Master +0military');
+                expect(this.warDogMaster.getMilitarySkill()).toBe(2);
+            });
+        });
+    });
```

Comment:
Might be worth having a test that it won't let you resolve it when your D deck is empty

Reply:
Good idea.  Done.